cargo-dist/src/tasks.rs

```diff
@@ -2795,3 +2810,30 @@ impl InstallReceipt {
         })
     }
 }
+
+// Determines the *cached* HEAD for a submodule within the workspace.
+// Note that any unstaged commits, and any local changes to commit
+// history that aren't reflected by the submodule commit history,
+// won't be reflected here.
+fn submodule_head(submodule_path: &Utf8PathBuf) -> DistResult<String> {
+    let output = Cmd::new("git", "fetch expected latest commit for a submodule")
+        .arg("submodule")
+        .arg("status")
+        .arg("--cached")
+        .arg(submodule_path)
+        .output()
+        .map_err(|_| DistError::GitSubmoduleCommitError {
+            path: submodule_path.to_string(),
```

Comment:
Should this wrap the axoprocess error? Oh I see, the other path doesn't have that.